doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -44,11 +44,14 @@ Rewriting with Leibniz and setoid equality
       oriented_rewriter ::= {? {| -> | <- } } {? @natural } {? {| ? | ! } } @one_term_with_bindings
       one_term_with_bindings ::= {? > } @one_term {? with @bindings }
 
-   Rewrites terms based on equalities.  The type of :n:`@one_term` must have the form:
+   Replace terms with other terms by using proofs that these are
+   equals.  The type of :n:`@one_term` must have the form:
```

Comment:
```suggestion
   Rewrite subterms with other subterms that have been proven to be equal.
     The type of :n:`@one_term` must have the form:
```

I prefer using "rewrite" in this first sentence because a) it's the name of the tactic being described and b) there is a distinct "replace" tactic.  The ordinary meanings of the two words seem almost synonymous.  Not clear to me we want to make precisely distinguished technical terms, either.  Though arguably "replace" is slightly clearer than "rewrite".  Up to you.

Reply:
My intuition is that we rewrite the entire goal by replacing subterms with other subterms. I'll keep the word "replace" here and apply the rest of your suggestion.